client.go

```diff
@@ -1375,3 +1397,545 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+			WriterID:     response.Metadata.WriterID.String(),
+			UserID:       response.Metadata.UserID.String(),
+			Type:         response.Metadata.Type,
+			Plain:        response.Metadata.Plain,
+			Created:      response.Metadata.Created,
+			LastModified: response.Metadata.LastModified,
+			Version:      response.Metadata.Version.String(),
+			FileMeta:     (*pdsClient.FileMeta)(response.Metadata.FileMeta),
+		},
+		Data:            response.Data,
+		RecordSignature: response.RecordSignature,
+	}
+	return fileSecret, nil
+}
+
+// WriteRecord encrypts the data for the record and creates a new record in E3DB
+func (c *ToznySDKV3) WriteRecord(ctx context.Context, data map[string]string, recordType string, plain map[string]string) (*pdsClient.Record, error) {
+	recordToWrite := pdsClient.WriteRecordRequest{
+		Data: data,
+		Metadata: pdsClient.Meta{
+			Type:     recordType,
+			WriterID: c.E3dbPDSClient.ClientID,
+			UserID:   c.E3dbPDSClient.ClientID,
+			Plain:    plain,
+		},
+	}
+	encryptedRecord, err := c.E3dbPDSClient.EncryptRecord(ctx, recordToWrite)
+	if err != nil {
+		return nil, err
+	}
+	recordSecret, err := c.E3dbPDSClient.WriteRecord(ctx, encryptedRecord)
+	if err != nil {
+		return nil, err
+	}
+	return recordSecret, nil
+}
+
+func (c *ToznySDKV3) ShareRecordWithGroup(ctx context.Context, recordType string, group *storageClient.Group) error {
+	keyReq := pdsClient.GetOrCreateAccessKeyRequest{
+		WriterID:   c.E3dbPDSClient.ClientID,
+		UserID:     c.E3dbPDSClient.ClientID,
+		ReaderID:   c.E3dbPDSClient.ClientID,
+		RecordType: recordType,
+	}
+	keyResp, err := c.E3dbPDSClient.GetOrCreateAccessKey(ctx, keyReq)
+	if err != nil {
+		return err
+	}
+	encryptionKeys := e3dbClients.EncryptionKeys{
+		Public: e3dbClients.Key{
+			Type:     e3dbClients.DefaultEncryptionKeyType,
+			Material: c.StorageClient.EncryptionKeys.Public.Material,
+		},
+		Private: c.StorageClient.EncryptionKeys.Private,
+	}
+	wrappedAK, err := e3dbClients.EncryptAccessKey(keyResp, encryptionKeys)
+	if err != nil {
+		return err
+	}
+	accessKeyReq := storageClient.GroupAccessKeyRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: wrappedAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, encryptedGroupAK, err := c.StorageClient.CreateGroupAccessKey(ctx, accessKeyReq)
+	if err != nil {
+		return err
+	}
+	// share secret with group
+	secretShareReq := storageClient.ShareGroupRecordRequest{
+		GroupID:            group.GroupID,
+		RecordType:         recordType,
+		EncryptedAccessKey: encryptedGroupAK,
+		PublicKey:          group.PublicKey,
+	}
+	_, err = c.StorageClient.ShareRecordWithGroup(ctx, secretShareReq)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetOrCreateNamespace creates the group for the namespace if it doesn't exist and returns the Group
+func (c *ToznySDKV3) GetOrCreateNamespace(ctx context.Context, realmName string, namespace string) (*storageClient.Group, error) {
```

Comment:
with the same argument that namespaces are likely to morph over time, probably a good idea to do some future proofing and collect the realmName and namespace into an options struct for this method.